### PR TITLE
New version 2023-05-01 for RP Microsoft.Consumption 

### PR DESCRIPTION
**Pull request by pednekarpraj**
In this PR, I am creating a new version 2023-05-01 for Consumption RP.
I am making minor change in the reservation recommendation details API in this PR. Swagger documentation for other APIs in this RP remains same as current public stable version 2023-03-01.

Explanation: In the latest version 2023-05-01, subscriptionId will be accepted via oData filter for billing account and billing profile paths.

### CODE REVIEW

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json [hunk 1]
```diff
@@ -0,0 +1,5893 @@
...
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/scopeReservationRecommendationDetailsParameter"
+          },
+          {
+            "$ref": "#/parameters/recommendationDetailsFilterParameter"
```
Comment (pednekarpraj) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json:1160]: New change in this version. Earlier in reservation recommendation details API, we used to accept query parameters like scope, region, term, lookback period and product. In this latest version 2023-05-01, these inputs will be accepted via oData filter.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json [hunk 2]
```diff
@@ -0,0 +1,5893 @@
...
+    },
+    "filterParameter": {
+      "name": "$filter",
+      "in": "query",
+      "description": "May be used to filter reservationRecommendations by: properties/scope with allowed values ['Single', 'Shared'] and default value 'Single'; properties/resourceType with allowed values ['VirtualMachines', 'SQLDatabases', 'PostgreSQL', 'ManagedDisk', 'MySQL', 'RedHat', 'MariaDB', 'RedisCache', 'CosmosDB', 'SqlDataWarehouse', 'SUSELinux', 'AppService', 'BlockBlob', 'AzureDataExplorer', 'VMwareCloudSimple'] and default value 'VirtualMachines'; and properties/lookBackPeriod with allowed values ['Last7Days', 'Last30Days', 'Last60Days'] and default value 'Last7Days'.",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "recommendationDetailsFilterParameter": {
```
Comment (pednekarpraj) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json:5867]: New change in this version. Earlier in reservation recommendation details API, we used to accept query parameters like scope, region, term, lookback period and product. In this latest version 2023-05-01, these inputs will be accepted via oData filter.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json (hunk 1 above)
Comment (ihsanali01) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json:1160]: Thankful for this opportunity and hope.

**Comment by zedy-wj** [comment]
@raych1 - Please help review the Go breaking change.

**Comment by zedy-wj** [comment]
@msyyc  - Please help review the Python breaking change.

**Comment by zedy-wj** [comment]
@qiaozha - Please help review JS breaking change.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption_pricesheet.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
+{
+    "swagger": "2.0",
+    "info": {
+      "version": "2023-03-01",
```
Comment (qiaozha) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption_pricesheet.json:4]: Is this suppose to be 2023-05-01 ?

**Review by live1206 — APPROVED** [review]

**Pull request merged**